src/legacy/core_plugins/kibana/public/visualize/np_ready/legacy_app.js

```diff
@@ -155,13 +162,17 @@ export function initVisualizeApp(app, deps) {
               })
               .catch(
                 redirectWhenMissing({
-                  visualization: '/visualize',
-                  search:
-                    '/management/kibana/objects/savedVisualizations/' + $route.current.params.id,
-                  'index-pattern':
-                    '/management/kibana/objects/savedVisualizations/' + $route.current.params.id,
-                  'index-pattern-field':
-                    '/management/kibana/objects/savedVisualizations/' + $route.current.params.id,
+                  history,
+                  mapping: {
+                    visualization: '/visualize',
+                    search:
```

Comment:
Those redirects don't work for me. Steps to reproduce:
* Create visualization based on index pattern
* Go to index pattern management and delete index pattern
* Go to Visualize listing page
* Click visualization
* Nothing happens (should redirect)

Same for the saved search case when the index pattern is missing

Reply:
Good catch. But seems this is broken for ages,
it tries to check the error is an instance of `SavedObjectNotFound`,
but there we have two copies of those class. The error is an instance of class from one bundle `http://localhost:5620/plugin:data/src\plugins\kibana_utils\public\errors\errors.ts`, but it tries to check the instance of the same class from other bundle: `http://localhost:5620/bundles/D:\kibana\src\plugins\kibana_utils\public\errors\errors.ts` 

![image](https://user-images.githubusercontent.com/31325372/76741501-b1854f00-6780-11ea-9a29-898e4d42db50.png)

So I changed the logic to check `savedObjectType` prop erty and left a comment there.

Reply:
Tested and works now 👍

Reply:
Could you pls check to following case:
The redirection to the management section works, but when I try to navigate back, click on visualize in the sidebar, I'm redirected to Management again

Reply:
if I don't delete the visualization in management

Reply:
left a note in #59999, which reports a similar behavior

Reply:
left a note in #59999, which reports a similar behavior